pkg/cloudmeta/aws.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2024 ScyllaDB
+
+package cloudmeta
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/pkg/errors"
+)
+
+// AWSMetadata is a wrapper around ec2 metadata client.
+type AWSMetadata struct {
+	ec2meta *ec2metadata.EC2Metadata
+}
+
+// NewAWSMetadata is a constructor for  AWSMetadata service.
+// testEndpoint can be provided if you want to overwrite the default metadata endpoint, otherwise leave it empty.
+func NewAWSMetadata(testEndpoint string) (*AWSMetadata, error) {
```

Comment:
What do you think about changing `testEndpoint` to just `endpoint`?
It can still be specified in the comment, that `endpoint` should be used mostly for testing.
It just looks strange to me to see a "test" argument in the regular code base directly.

Or perhaps it would be nicer to have a separate method for `NewAWSMetadataWithEndpoint` which would share most of the body with `NewAwsMetadata`?

No strong opinions on that.

Reply:
yes, make sense, it may look confusing without having actual test in the commit :) 
I'll update this part.

Reply:
I would suggest to allow to override the whole aws config instead of just endpoint.